llvm/include/llvm/CodeGen/RegisterPressure.h

```diff
@@ -35,11 +35,11 @@ class MachineInstr;
 class MachineRegisterInfo;
 class RegisterClassInfo;
 
-struct RegisterMaskPair {
+struct RegUnitMaskPair {
   Register RegUnit; ///< Virtual register or register unit.
   LaneBitmask LaneMask;
 
-  RegisterMaskPair(Register RegUnit, LaneBitmask LaneMask)
+  RegUnitMaskPair(Register RegUnit, LaneBitmask LaneMask)
       : RegUnit(RegUnit), LaneMask(LaneMask) {}
```

Comment:
This is really something like `std::variant<unsigned RegUnit, std::pair<Register, LaneBitmask>>`.

Maybe LiveInReg? That's all this is used for I believe

Reply:
It seems to be used throughout the RegisterPressure code for more than just live ins.